### PR TITLE
Fix sonar code smell issues

### PR DESCRIPTION
**Pull request by mreis-ch**
- Rename class members that have the same name as their classes Costs, Error and Errors. Update getters and setters and classes that call these.
- Reduce verbosity of generics in AccountsNoteConverter
- Use primitive boolean in FilingServiceImpl
- The majority of changes are the removal of the public modifier from unit tests and methods within.
- Removed any unused imports or variables which were not used.
- Added, where possible, parameterised generics

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
     @JsonProperty("errors")
     @Field("errors")
-    private Set<Error> errors = new HashSet<>();
+    private Set<Error> errorsList = new HashSet<>();
```
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java:16]: Same as the previous comment, could this cause issues with spring?

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java [hunk 2]
```diff
@@ -6,14 +6,14 @@
 public class Costs {
 
     @JsonProperty("costs")
-    private Map<String, Cost> costs;
+    private Map<String, Cost> costsList;
```
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:9]: Can this cause issue with spring-deserialising? usually the name of the field has to match the json property although i could be wrong (worth checking this)

File: src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java [hunk 3]
```diff
@@ -9,25 +9,25 @@
 
 import uk.gov.companieshouse.environment.EnvironmentReader;
 
-public class ApplicationConfigurationTest {
+class ApplicationConfigurationTest {
```
Comment (amoriartyCH) [inline src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java:12]: Do we know the default if public isn't specified? (Was this private before?)

**Review by amoriartyCH — COMMENTED** [review]
Looks good, before approving wanted to ask if we are okay with all of these changes going into 1 very large PR? As this will involve a lot of different testing scenarios for one ticket without the ability to test each change in isolation?

File: src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java (hunk 3 above)
Comment (mreis-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java:12]: From what I can see it's package-private

Comment (amoriartyCH) [reply src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java:12]: awesome

**Review by amoriartyCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java (hunk 2 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:9]: The point of JsonProperty is to rename a property.
https://github.com/FasterXML/jackson-annotations#annotations-for-renaming-properties

File: src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java (hunk 1 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java:16]: I don't think so. See previous answer.

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java (hunk 2 above)
Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:9]: yeah, investigated this and talked about it in slack, all good 👍

Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:9]: Would costsMap be better here? It is not a list.

File: src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java (hunk 1 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/api/accounts/model/validation/Errors.java:16]: errorSet?

File: src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java (hunk 3 above)
Comment (ch-tstrickland) [reply src/test/java/uk/gov/companieshouse/api/accounts/ApplicationConfigurationTest.java:12]: Yes, package private by default if unspecified in Java.

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java (hunk 2 above)
Comment (mreis-ch) [reply src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:9]: That's a fair point. I'll update

**Review by ch-tstrickland — COMMENTED** [review]
Minor comments aside, looks good to me.

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java [hunk 4]
```diff
@@ -6,14 +6,14 @@
...
+    private Map<String, Cost> costsMap;
 
-    public Map<String, Cost> getCosts() {
-        return costs;
+    public Map<String, Cost> getCostsList() {
+        return costsMap;
     }
 
-    public void setCosts(
+    public void setCostsList(
```
Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:15]: setCostsMap

File: src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java [hunk 5]
```diff
@@ -6,14 +6,14 @@
 public class Costs {
 
     @JsonProperty("costs")
-    private Map<String, Cost> costs;
+    private Map<String, Cost> costsMap;
 
-    public Map<String, Cost> getCosts() {
-        return costs;
+    public Map<String, Cost> getCostsList() {
```
Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/rest/Costs.java:11]: getCostsMap

**Review by ch-tstrickland — APPROVED** [review]

**Pull request merged**